recirq/otoc/parallel_xeb_example.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example for performing parallel-XEB and calibrating unitary shifts."""
+
+import cirq
+import numpy as np
+
+from recirq.otoc.parallel_xeb import build_xeb_circuits, parallel_xeb_fidelities
+from recirq.otoc.utils import save_data
+
+
+def main():
+    # Specify a working directory, project ID and processor name.
+    dir_str = ''
+    project_id = ''
+    engine = cirq.google.Engine(project_id=project_id,
+                                proto_version=cirq.google.ProtoVersion.V2)
+    processor_name = 'rainbow'
+
+    # Specify qubits to measure. Here we choose the qubits to be on a line.
+    qubit_locs = [(3, 3), (2, 3), (2, 4), (2, 5), (1, 5), (0, 5), (0, 6),
+                  (1, 6)]
+    qubits = [cirq.GridQubit(*idx) for idx in qubit_locs]
+    num_qubits = len(qubits)
+
+    # Specify the gate layers to calibrate. For qubits on a line, all two-qubit
+    # gates can be calibrated in two layers.
+    int_layers = [
+        {(qubit_locs[i], qubit_locs[i + 1]) for i in
+         range(0, num_qubits - 1, 2)},
+        {(qubit_locs[i], qubit_locs[i + 1]) for i in
+         range(1, num_qubits - 1, 2)}]
+
+    xeb_configs = [[cirq.Moment([cirq.ISWAP(qubits[i], qubits[i + 1]) ** 0.5
+                                 for i in range(0, num_qubits - 1, 2)])],
+                   [cirq.Moment([cirq.ISWAP(qubits[i], qubits[i + 1]) ** 0.5
+                                 for i in range(1, num_qubits - 1, 2)])]]
+
+    # Specify the number of random circuits in parallel XEB the cycles at which
+    # fidelities are to be measured.
+    num_trials = 20
+    num_cycle_range = range(3, 75, 10)
+    num_cycles = len(num_cycle_range)
+
+    # Generate random XEB circuits, measure the resulting bit-strings, and save
+    # the data as well as random circuit information.
+    all_bits_sqrtiswap = []
+    all_phases_sqrtiswap = []
+    for xeb_config in xeb_configs:
+        bits_sqrtiswap = []
+        phases_sqrtiswap = []
```

Comment:
why is this called `phases`? The return variable is named `rand_indices`. The second return value is not documented in `build_xeb_circuits`.

Reply:
The second return value is now documented. I also renamed these variables to be more intuitive.